### ISSUE TITLE
Document Date field

### ISSUE THREAD

**Issue by cosorio** [documentcloud/documentcloud#217]
Dear document cloud team,
We are using your API to build a search engine for our website visitors to search our documents in the DocumentCloud. 
We want to give the user the ability to search our documents by text, date, subject, and many other meta data we add to each document.
The problem is that there are not date fields currently that we could use, neither there is a "search by" date field or "sort by" date field in your API.

We definitely need to give the user the ability to search by the date of our documents (note that it is not the created at date, rather a date attribute for each document). The date attribute is a key research tool to organize the documents chronologically or narrow a search to a specific period. All researchers and institutions will benefit from this added “Document Date” research tool.
We are planning to put all our documents in DocumentCloud by the end of the summer. We currently have more than 14000 documents. We need your help in order to be able to use your service. To do that, we need you to please add a “Document Date” field in DocumentCloud, and allow the API to search by date and sort by date.  

Please let us know when this is possible.

Thank you,

The National Security Archive team

**Comment by jashkenas** [comment]
Hey Carlos.

That's very exciting news.

DocumentCloud doesn't come with a pre-baked set of metadata attributes for every organization. Rather, it allows you to define your own metadata attributes using whatever schema fits your archive best.

Read more about it under the **Editing and Searching your own Custom Data** section at the bottom of this page: https://www.documentcloud.org/help/searching

In short, you can add "date" and "subject", and many other metadata fields to your document sets via the API, and use them to filter in conjunction with the full text search.

That said, I don't believe there's currently support for fancier operators like "before x date" or "after y date" yet — if you need that sort of thing, there's more work that will need to be done.

**Comment by cosorio** [comment]
Dear Jeremy,

 

We at the archive are about to migrate two collections (some30,000 documents) to document cloud. We understand very well that we can add tailor made fields to the collections as you explain in your last email. But the disappointing thing is that one cannot sort records by those fields, nor can one search ranges of dates.

 

We still cannot understand why there is not an administrative  “Document Date” field to sort documents or to search within a range. This is a very basic administrative and research tool that is part of documents databases online:

 

https://foia.state.gov/Search/Search.aspx

 

http://aad.archives.gov/aad/fielded-search.jsp?dt=2472 <http://aad.archives.gov/aad/fielded-search.jsp?dt=2472&cat=WR43&tf=X&bc=,sl> &cat=WR43&tf=X&bc=,sl

 

Do you know how we could help encourage designing and deploying such capability?

 

 

Carlos
